coderd/database/migrations/000282_user_status_changes.up.sql

```diff
@@ -0,0 +1,65 @@
+CREATE TABLE user_status_changes (
+    id uuid PRIMARY KEY DEFAULT gen_random_uuid(),
+    user_id uuid NOT NULL REFERENCES users(id),
+    new_status user_status NOT NULL,
+    changed_at timestamptz NOT NULL DEFAULT CURRENT_TIMESTAMP
+);
+
+COMMENT ON TABLE user_status_changes IS 'Tracks the history of user status changes';
+
+CREATE INDEX idx_user_status_changes_changed_at ON user_status_changes(changed_at);
+
+INSERT INTO user_status_changes (
+    user_id,
+    new_status,
+    changed_at
+)
+SELECT
+    id,
+    status,
+    created_at
+FROM users
+WHERE NOT deleted;
+
+CREATE TABLE user_deleted (
+    id uuid PRIMARY KEY DEFAULT gen_random_uuid(),
+    user_id uuid NOT NULL REFERENCES users(id),
+    deleted_at timestamptz NOT NULL DEFAULT CURRENT_TIMESTAMP
+);
```

Comment:
Will we ever be able to "un-delete" a user? Would is be easy to implement this to handle that case if we ever support it?

Reply:
That is true. I assume we can still make an index with a conditional expression on `status == deleted`. We might want to have some trigger or something to prevent "un-deletion" if we want to prevent that.

Reply:
I considered these options, but avoided them simply because they expand the scope beyond a reasonable time cost for this feature. We could relatively easy migrate from this to a space where `deleted` is just another user status.

I'm trying to remain agnostic of the massive rabbit-hole shaped question of what the business rule should be around un/re-deletion.

The trigger and function right now would handle un/re-deletion correctly, but the query would break as identified by @Emyrk in another comment below.

I would prefer that we stick to the current implementation because it is good enough and I can't find anywhere that we support un/re-deletion right now. If y'all would like to explicitly request a specific change I'd be happy to consider it.